### ISSUE TITLE
Method 1: Threshold optimization (literature + data-driven)

### ISSUE THREAD

**Issue by ofekvs18** [ofekvs18/UnderGradProject#2]
## Context

This is the first biomarker generation method in the framework. We have a verified, leakage-free pipeline for Rheumatoid Arthritis prediction from CBC data.

**Baseline to beat**: Logistic regression on all 9 CBC features → AUC 0.658

**Data location**: `data/ra_modeling_data.csv`
- Columns: subject_id, is_case, split, hct, hgb, mch, mchc, mcv, plt, rbc, rdw, wbc
- Split column is pre-computed and immutable. Use it as-is.

## Goal

Implement the simplest possible biomarker: a single CBC feature with a threshold.
- "If RDW > X, predict RA" is a biomarker.
- Compare two approaches to choosing the threshold:
  - **1A**: From published clinical literature
  - **1B**: Optimized from training data (Youden's index)

This establishes the single-feature baseline that multi-feature methods (random formulas, GP, LLM) must beat.

#### Part 1A: Literature-based thresholds

For each CBC feature, apply known clinical thresholds associated with inflammatory/autoimmune conditions. Use these values (from standard clinical references):

```python
literature_thresholds = {
    # feature: (threshold, direction, source)
    "rdw":  (14.5, "above", "RDW >14.5% associated with inflammatory conditions"),
    "hgb":  (12.0, "below", "Anemia (Hgb <12 g/dL) common in chronic inflammation"),
    "hct":  (36.0, "below", "Low hematocrit in anemia of chronic disease"),
    "wbc":  (11.0, "above", "Leukocytosis in active inflammation"),
    "plt":  (400.0, "above", "Thrombocytosis (reactive) in chronic inflammation"),
    "mcv":  (80.0, "below", "Microcytosis in anemia of chronic disease"),
    "mch":  (27.0, "below", "Low MCH in iron-deficiency/chronic disease anemia"),
    "mchc": (32.0, "below", "Low MCHC in chronic disease"),
    "rbc":  (4.0, "below", "Low RBC in anemia of chronic disease"),
}
```

For each feature:
1. Apply the threshold to the TEST set (not train)
2. "above" means predict RA if feature > threshold
3. "below" means predict RA if feature < threshold
4. Compute: AUC (using the raw feature value, not the binary prediction), precision, recall, F1 at the literature threshold

#### Part 1B: Data-driven thresholds (Youden's index)

For each CBC feature:
1. Using only the TRAIN set, compute ROC curve
2. Find optimal threshold via Youden's index: maximize (sensitivity + specificity - 1)
3. Apply that threshold to the TEST set
4. Compute: AUC, precision, recall, F1 at the optimal threshold
5. Also record what the optimal threshold value is

#### Part 1C: Comparison

Create a comparison table and visualizations:
1. Table: feature, literature_threshold, literature_precision, literature_recall, datadriven_threshold, datadriven_precision, datadriven_recall, AUC
2. Bar chart: AUC by feature (grouped: literature vs data-driven)
3. For the best feature (highest AUC), plot both thresholds on the ROC curve
4. Save all outputs to `results/method1_threshold/`

### Output files

```
results/method1_threshold/
├── literature_results.csv
├── datadriven_results.csv
├── comparison_table.csv
├── comparison_chart.png
├── best_feature_roc.png
└── method1_summary.txt
```

### Summary output format (method1_summary.txt)

```
Method 1: Threshold Optimization Results
=========================================

Best single feature: [feature name]
  Literature threshold: [value] ([direction]) → AUC=[x], Precision=[x], Recall=[x]
  Data-driven threshold: [value] ([direction]) → AUC=[x], Precision=[x], Recall=[x]
  Improvement from data-driven: [+/- x AUC]

Comparison to baseline:
  All-features logistic regression AUC: 0.658
  Best threshold AUC: [x]
  Difference: [+/- x]

Key insight: [Does data-driven meaningfully beat literature? By how much?]
```

### Dependencies

- pandas
- scikit-learn
- matplotlib
- numpy

### Code style

- One file, no classes, functions only
- Print progress as it runs
- Load data once at the top
- Use the pre-computed split column (never re-split)
- Comment the Youden's index formula

### What NOT to do

- Do NOT tune anything beyond the threshold
- Do NOT combine features (that is Method 2+)
- Do NOT use the test set for threshold selection (only Youden on train)
- Do NOT build abstractions or frameworks
